R/classes.R

```diff
@@ -26,29 +26,23 @@ create_model_object <- function(res, save_yaml) {
     strict_mode_error(err_msg)
   }
 
-  # check model type
+  # assign class and write YAML to disk
   .model_type <- res[[YAML_MOD_TYPE]]
   if (!(.model_type %in% SUPPORTED_MOD_TYPES)) {
-    stop(glue("Invalid {YAML_MOD_TYPE} `{.model_type}`. Valid options include: `{paste(SUPPORTED_MOD_TYPES, collapse = ', ')}`"))
+    dev_error(glue("Invalid {YAML_MOD_TYPE} `{.model_type}`. Valid options include: `{paste(SUPPORTED_MOD_TYPES, collapse = ', ')}`"))
   }
 
-  # we won't know the model file extension, so we rely on this helper to check
-  # the possible extensions and throw an error if none exists
-  find_nonmem_model_file_path(res[[ABS_MOD_PATH]], .check_exists = TRUE)
+  # assign class
+  class(res) <- c(as.character(glue("bbi_{.model_type}_model")), BBI_PARENT_CLASS, class(res))
 
-  # check bbi args and add an empty list if missing
-  if (is.null(res[[YAML_BBI_ARGS]])) {
-    res[[YAML_BBI_ARGS]] <- list()
-  } else {
-    # check that unique named list was passed
-    tryCatch(
-      checkmate::assert_list(res[[YAML_BBI_ARGS]], names="unique"),
-      error = function(e) { stop(glue("`{YAML_BBI_ARGS}` must be a unique, named list: {e}")) }
-    )
+  # look for appropriate model files on disk and then write out YAML.
+  # must be done AFTER assigning the class so that associated helpers can dispatch correctly
```

Comment:
This comment is somewhat unclear, because on literally the next line we check `.model_type` rather than the class of `res`. I think this mostly relates to `get_yaml_path()`, which is called by `save_model_yaml()`?